device/send.go

```diff
@@ -305,26 +349,55 @@ top:
 	}
 
 	for {
+		var elemsOOO *[]*QueueOutboundElement
 		select {
-		case elem := <-peer.queue.staged:
-			elem.peer = peer
-			elem.nonce = keypair.sendNonce.Add(1) - 1
-			if elem.nonce >= RejectAfterMessages {
-				keypair.sendNonce.Store(RejectAfterMessages)
-				peer.StagePacket(elem) // XXX: Out of order, but we can't front-load go chans
-				goto top
+		case elems := <-peer.queue.staged:
+			i := 0
+			for _, elem := range *elems {
+				elem.peer = peer
+				elem.nonce = keypair.sendNonce.Add(1) - 1
+				if elem.nonce >= RejectAfterMessages {
+					keypair.sendNonce.Store(RejectAfterMessages)
+					if elemsOOO == nil {
+						elemsOOO = peer.device.GetOutboundElementsSlice()
+					}
+					*elemsOOO = append(*elemsOOO, elem)
+					continue
+				} else {
+					(*elems)[i] = elem
+					i++
+				}
+
+				elem.keypair = keypair
+				elem.Lock()
 			}
+			*elems = (*elems)[:i]
 
-			elem.keypair = keypair
-			elem.Lock()
+			if elemsOOO != nil {
+				peer.StagePackets(elemsOOO) // XXX: Out of order, but we can't front-load go chans
+			}
+
+			if len(*elems) < 1 {
```

Comment:
`== 0`

Reply:
Updated in https://github.com/WireGuard/wireguard-go/pull/64/commits/3027b62522d027cf3198f6908a061337aa23c45f.